mailbox/api/src/main/java/org/apache/james/mailbox/quota/MaxQuotaManager.java

```diff
@@ -116,6 +116,13 @@
      */
     Optional<QuotaSize> getMaxStorage(QuotaRoot quotaRoot) throws MailboxException;
 
+    default Optional<QuotaSize> getMaxStorage(Map<Quota.Scope, QuotaSize> maxStorageDetails) {
```

Comment:
Can't the implems (Cassandra, JPA etc...) call these methods?

That would guaranty that we compute quota in a single place, regardless of the implementation....